spec/lib/audit/catalog_to_moab_spec.rb

```diff
@@ -108,6 +108,30 @@
       described_class.send(:check_catalog_version, pres_copy, nil)
     end
 
+    it 'calls .moab_found_on_disk?(moab,results)' do
+      moab = instance_double(Moab::StorageObject)
+      allow(Moab::StorageObject).to receive(:new).with(druid, instance_of(String)).and_return(moab)
+      pohandler_results = instance_double(PreservedObjectHandlerResults)
+      allow(PreservedObjectHandlerResults).to receive(:new).and_return(pohandler_results)
+      expect(described_class).to receive(:moab_found_on_disk?).with(moab, pohandler_results)
+      described_class.send(:check_catalog_version, pres_copy, nil)
+    end
+
+    context 'moab is nil (exists in catalog but not on disk)' do
+      it 'adds an ONLINE_MOAB_DOES_NOT_EXIST result' do
+        allow(Moab::StorageObject).to receive(:new).with(druid, instance_of(String)).and_return(nil)
+        pohandler_results = instance_double(PreservedObjectHandlerResults, report_results: nil)
+        allow(PreservedObjectHandlerResults).to receive(:new).and_return(pohandler_results)
+        expect(pohandler_results).to receive(:add_result).with(
+          PreservedObjectHandlerResults::ONLINE_MOAB_DOES_NOT_EXIST
+        )
+        described_class.send(:check_catalog_version, pres_copy, nil)
+      end
+      it 'stops processing .check_catalog_version' do
+        skip 'add this test once we add more code to .check_catalog_version???'
```

Comment:
for this, right now you could do  

```ruby
expect(moab_double_object).not_to receive(:current_version_id)
```

because that is the next statement, and it is executed regardless of which comparison case occurs.